### PR TITLE
Crud for country entity (frontend)

### LINKED ISSUE #476: crud for Country entity (frontend)
Based on issue #451 implement a layout with crud functionality for **Country** entity.

### PR DESCRIPTION
**Pull request by dpmarkov**
The task was to create a **CRUD** functionality for **Country** entity.
The **layout** is based on the provided **Figma design**.

## Video Demo:

https://user-images.githubusercontent.com/71282381/154966986-205bebfc-520d-4baa-beff-a9f4c8bd45a7.mp4

close #476

### CODE REVIEW

File: src/components/countries/CountryForm.tsx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    CountryInput
+  >({
+    mutationFn: useCreateCountry(),
+    onError: () => AlertStore.show(t('alerts.new-row.error'), 'error'),
+    onSuccess: () => AlertStore.show(t('alerts.new-row.success'), 'success'),
+  })
+  type CountryEditInput = {
+    id: string
+    data: CountryInput
+  }
```
Comment (kachar) [inline src/components/countries/CountryForm.tsx:60]: It's a good practice to define the types outside of the function body so they can be exported or reused if needed

```suggestion
type CountryFormProps = {
  initialValues?: CountryInput
  id?: string
}
type CountryEditInput = {
  id: string
  data: CountryInput
}

export default function CountryForm({ initialValues = defaults, id }: CountryFormProps) {
  //if (id) -> edit form, else -> create form
  const { t } = useTranslation('countries')

  const createMutation = useMutation<
    AxiosResponse<CountryResponse>,
    AxiosError<ApiErrors>,
    CountryInput
  >({
    mutationFn: useCreateCountry(),
    onError: () => AlertStore.show(t('alerts.new-row.error'), 'error'),
    onSuccess: () => AlertStore.show(t('alerts.new-row.success'), 'success'),
  })
```

File: src/components/countries/CountryForm.tsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+  const classes = useStyles()
+
+  const onSubmit = async (
+    values: CountryInput,
+    { setFieldError, resetForm }: FormikHelpers<CountryInput>,
+  ) => {
+    try {
+      const data = {
+        name: values.name,
+        countryCode: values.countryCode.toLocaleUpperCase(),
```
Comment (kachar) [inline src/components/countries/CountryForm.tsx:80]: Good call to uppercase the country code :raised_hands:

File: src/pages/admin/countries/[id]/index.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import { dehydrate, QueryClient } from 'react-query'
+import { serverSideTranslations } from 'next-i18next/serverSideTranslations'
+
+import { queryFn } from 'service/restRequests'
+import CountryEditPage from 'components/countries/CountryEditPage'
+
+export const getServerSideProps: GetServerSideProps = async ({ locale, query }) => {
+  const { id } = query
+  const client = new QueryClient()
+  await client.prefetchQuery(`/country/${id}`, queryFn)
```
Comment (kachar) [inline src/pages/admin/countries/[id]/index.tsx:11]: Please use the route from api endpoints so it's defined in a single location

**Review by kachar — COMMENTED** [review]
Well done @dpmarkov :medal_military: 

Thanks for providing a video of the interaction with the UI, really helpful.

There are some minor things to update before we merge this PR.

I saw many good practices in the PR, good job.

File: src/components/countries/CountryForm.tsx (hunk 2 above)
Comment (dpmarkov) [reply src/components/countries/CountryForm.tsx:80]: I probably should of done that on the back-end too.

Comment (kachar) [reply src/components/countries/CountryForm.tsx:82]: Good idea

**Review by kachar — APPROVED** [review]
:clap:

**Comment by Polina1985** [comment]
I have tested as an administrator and I:
- have created a Country entry and country code without a problem.
@dpmarkov and @kachar  
I am moving this ticket to DONE

**Pull request merged**